src/mdo/java/ImmutableCollections.java

```diff
@@ -79,32 +65,7 @@ public int size() {
     };
 
     static <E> List<E> copy(Collection<E> collection) {
-        if (collection == null) {
-            return emptyList();
-        } else if (collection instanceof AbstractImmutableList) {
-            return (List<E>) collection;
-        } else {
-            switch (collection.size()) {
-                case 0:
-                    return emptyList();
-                case 1:
-                    return singletonList(collection.iterator().next());
-                case 2:
-                    Iterator<E> it = collection.iterator();
-                    return new List2<>(it.next(), it.next());
-                default:
-                    return new ListN<>(collection);
-            }
-        }
-    }
-
-    @SuppressWarnings("unchecked")
-    static <E> List<E> emptyList() {
-        return (List<E>) EMPTY_LIST;
-    }
-
-    static <E> List<E> singletonList(E element) {
-        return new List1<>(element);
+        return collection == null ? List.of() : List.copyOf(collection);
```

Comment:
Why not `Collections.emptyList()`? Does the empty list needs to be mutable?

Reply:
List.of() also returns an immutable list (although not the same objectas Collections.emptyList()).  
It's a more concise form, and I think the best practice will be to use List.of() and Map.of() instead of the similar methods from Collections.

Reply:
another advantage of using `Foo.of()` is that any defensive `Foo.copyOf()` downstream becomes a no-op.

https://github.com/openjdk/jdk/blob/a300c356555019a42c19bf0c16184f6dee4ad96e/src/java.base/share/classes/java/util/ImmutableCollections.java#L186-L187